portal/common/lib/types/index.ts

```diff
@@ -25,50 +25,68 @@ export type Resource = {
     headers: Map<string, string>;
     blob_id: string;
     blob_hash: string;
+    range: Range | null;
 };
 
+export type Range = {
+    start: number | null;
+    end: number | null;
+};
+
+function rangeToHttpHeader(range: Range): string {
+    return `bytes=${range.start || ""}-${range.end || ""}`;
+}
+
+export function optionalRangeToHeaders(range: Range | null): { [key: string]: string } {
+    if (range) {
+        return { range: rangeToHttpHeader(range) };
+    } else {
+        return {};
+    }
+}
+
 export type VersionedResource = Resource & {
     version: string; // the sui object version of the site resource
     objectId: string; // the sui object id of the site resource
 };
 
 /**
  * Type guard for the Resource type.
-*/
+ */
 export function isResource(obj: any): obj is Resource {
     return (
         obj &&
-        typeof obj.path === 'string' &&
-        typeof obj.headers === 'object' &&
-        typeof obj.blob_id === 'string' &&
-        typeof obj.blob_hash === 'string'
+        typeof obj.path === "string" &&
+        typeof obj.headers === "object" &&
+        typeof obj.blob_id === "string" &&
+        typeof obj.blob_hash === "string" &&
+        typeof obj.range === "object"
```

Comment:
Note: we should fix the editor config to include one of either single or double quotes. 
This PR looks big because there are multiple cases like this 😅

Reply:
i think the issue is that `pre-commit` was not run at times, and therefore the previous code was not consistently formatted with that. Agree that we should enforce this programmatically.